### PR TITLE
pd: add comments

### PR DESCRIPTION
**Pull request by Hijiao**
Signed-off-by: Max <xiahaijiao@hotmail.com>

## What have you changed? (mandatory)

Improve comment for src/pd

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No need
## Does this PR affect documentation (docs) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

File: src/pd/mod.rs [hunk 1]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
 
 pub const INVALID_ID: u64 = 0;
 
-// Client to communicate with placement driver (pd) for special cluster.
-// Because now one pd only supports one cluster, so it is no need to pass
-// cluster id in trait interface every time, so passing the cluster id when
-// creating the PdClient is enough and the PdClient will use this cluster id
-// all the time.
+/// PdClient communicates with placement driver (PD) for special cluster.
```
Comment (huachaohuang) [inline src/pd/mod.rs]: What does "special cluster" mean here? Maybe just remove that part.

```suggestion
/// PdClient communicates with Placement Driver (PD).
```

File: src/pd/mod.rs [hunk 2]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
...
     }
 
-    // Get cluster meta information.
+    /// Gets cluster meta information.
     fn get_cluster_config(&self) -> Result<metapb::Cluster>;
 
-    // For route.
-    // Get region which the key belong to.
+    /// For route.
+    /// Gets Region which the key belong to.
```
Comment (CaitinChen) [inline src/pd/mod.rs]:
```suggestion
    /// Gets Region which the key belongs to.
```

File: src/pd/mod.rs [hunk 3]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
...
     fn get_region(&self, key: &[u8]) -> Result<metapb::Region>;
 
-    // Get region info which the key belong to.
+    /// Gets Region info which the key belong to.
```
Comment (CaitinChen) [inline src/pd/mod.rs]: ditto

File: src/pd/mod.rs [hunk 4]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
...
-    // Please note that this method should only be called once.
+    /// Gets a stream of Region heartbeat response.
+    ///
+    /// Please note that this method should only be called once.
     fn handle_region_heartbeat_response<F>(&self, store_id: u64, f: F) -> PdFuture<()>
     where
         F: Fn(pdpb::RegionHeartbeatResponse) + Send + 'static;
 
-    // Ask pd for split, pd will returns the new split region id.
+    /// Asks PD for split, PD returns the new split Region id.
```
Comment (CaitinChen) [inline src/pd/mod.rs]:
```suggestion
    /// Asks PD for split. PD returns the newly split Region id.
```

File: src/pd/mod.rs [hunk 5]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
...
     fn ask_split(&self, region: metapb::Region) -> PdFuture<pdpb::AskSplitResponse>;
 
-    // Ask pd for batch split, pd will returns the new split region ids.
+    /// Asks PD for batch split, PD returns the new split Region ids.
```
Comment (CaitinChen) [inline src/pd/mod.rs]: ditto

File: src/pd/mod.rs [hunk 6]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
...
-    // Scatter the region across the cluster.
+    /// Scatters the Region across the cluster.
     fn scatter_region(&self, _: RegionInfo) -> Result<()> {
         unimplemented!();
     }
 
-    // Register a handler to the client, it will be invoked after reconnecting to PD.
-    //
-    // Please note that this method should only be called once.
+    /// Registers a handler to the client, it will be invoked after reconnecting to PD.
```
Comment (CaitinChen) [inline src/pd/mod.rs]:
```suggestion
    /// Registers a handler to the client, which will be invoked after reconnecting to PD.
```

File: src/pd/pd.rs [hunk 7]
```diff
@@ -45,21 +45,21 @@ use util::time::time_now_sec;
...
 
-// Use an asynchronous thread to tell pd something.
+/// Uses an asynchronous thread to tell PD something.
 pub enum Task {
     AskSplit {
         region: metapb::Region,
         split_key: Vec<u8>,
         peer: metapb::Peer,
-        // If true, right region derive origin region_id.
+        // If true, right Region derive origin region_id.
```
Comment (CaitinChen) [inline src/pd/pd.rs]:
```suggestion
        // If true, right Region derives origin region_id.
```

File: src/pd/pd.rs [hunk 8]
```diff
@@ -45,21 +45,21 @@ use util::time::time_now_sec;
...
         right_derive: bool,
         callback: Callback,
     },
     AskBatchSplit {
         region: metapb::Region,
         split_keys: Vec<Vec<u8>>,
         peer: metapb::Peer,
-        // If true, right region derive origin region_id.
+        // If true, right Region derive origin region_id.
```
Comment (CaitinChen) [inline src/pd/pd.rs]: ditto

File: src/pd/mod.rs [hunk 9]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
...
-    // It may happen that multi nodes start at same time to try to
-    // bootstrap, and only one can success, others will fail
-    // and must remove their created local region data themselves.
+    /// Creates the cluster with cluster ID, node, stores and first Region.
+    /// If the cluster is already bootstrapped, return ClusterBootstrapped error.
+    /// When a node starts, if it finds nothing in the node and
+    /// cluster is not bootstrapped, it begins to create node, stores, first Region
+    /// and then call bootstrap_cluster to let PD know it.
+    /// It may happen that multi nodes start at same time to try to
+    /// bootstrap, and only one can success, others will fail
```
Comment (CaitinChen) [inline src/pd/mod.rs]:
```suggestion
    /// bootstrap, but only one can succeed, while others will fail
```

File: src/pd/mod.rs [hunk 10]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
...
-    // and peers will be removed.
-    // When doing auto-balance, pd determines how to move the region from one store to another.
-
-    // Get store information.
+    /// We don't need to support Region and Peer put/delete,
+    /// because PD knows all Region and Peers itself.
+    /// When bootstrapping, PD knows first Region with `bootstrap_cluster`.
+    /// When changing Peer, PD determines where to add a new Peer in some store
+    /// for this Region.
+    /// When Region splitting, PD determines the new Region id and Peer id for the
```
Comment (CaitinChen) [inline src/pd/mod.rs]:
```suggestion
    /// When Region is splitting, PD determines the new Region id and Peer id for the
```

File: src/pd/mod.rs [hunk 11]
```diff
@@ -69,112 +69,112 @@ impl Deref for RegionInfo {
...
+    /// split Region.
+    /// When Region merging, PD knows which two Regions will be merged and which Region
```
Comment (CaitinChen) [inline src/pd/mod.rs]:
```suggestion
    /// When Region is merging, PD knows which two Regions will be merged and which Region
```

**Comment by Hijiao** [comment]
@CaitinChen Thanks, I addressed your comments, please take a look again.

**Pull request merged**